src/plugins/console/server/routes/api/console/autocomplete_entities/register_get_route.ts

```diff
@@ -17,25 +19,42 @@ interface Settings {
   dataStreams: boolean;
 }
 
+const RESPONSE_SIZE_LIMIT = 10 * 1024 * 1024;
+
+async function limitEntityResponseSize(esClient: IScopedClusterClient, entity: string) {
+  const stream = await esClient.asInternalUser.transport.request(
+    {
+      method: 'GET',
+      path: `/${entity}`,
+    },
+    { asStream: true }
+  );
+
+  // Limit the response size to 10MB, because the response can be very large and sending it to the client
+  // can cause the browser to hang.
+  const body = await streamToString(stream as unknown as IncomingMessage, RESPONSE_SIZE_LIMIT);
+  return JSON.parse(body);
+}
+
 async function getMappings(esClient: IScopedClusterClient, settings: Settings) {
   if (settings.fields) {
-    return esClient.asInternalUser.indices.getMapping();
+    return limitEntityResponseSize(esClient, '_mapping');
```

Comment:
I remember I've already asked you about why we cannot use 'maxResponseSize', but should be please answer again.... On my local it works 

<img width="684" alt="image" src="https://user-images.githubusercontent.com/20072247/196205405-8a6c9359-8aff-4a75-8ee4-f0a5f5244547.png">


<img width="867" alt="image" src="https://user-images.githubusercontent.com/20072247/196205153-e4ab36a0-0a1e-4f57-95e7-dea8ba0b4ab8.png">

Reply:
Looks like it determines the size of the response by looking at the `content-length` header, which may not always be set and may not be an ideal solution. See https://github.com/elastic/elastic-transport-js/blob/main/src/connection/HttpConnection.ts#L146